### PR TITLE
Some setter improve

### PR DESCRIPTION
**Pull request by minggo**
Re: https://github.com/cocos/cocos-engine/issues/16952

### Changelog

* only update rotation when the value is changed
* only update position when the value is changed

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/scene-graph/node.ts [hunk 1]
```diff
@@ -2106,17 +2106,27 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
...
+
         if (y === undefined && z === undefined) {
-            Vec3.copy(this._lpos, val as Vec3);
+            x = (val as Vec3).x;
+            y = (val as Vec3).y;
+            z = (val as Vec3).z;
         } else if (z === undefined) {
-            Vec3.set(this._lpos, val as number, y!, this._lpos.z);
+            x = val as number;
+            z = localPosition.z;
```
Comment (dumganhar) [inline cocos/scene-graph/node.ts:2118]: `y` is missing?

File: cocos/scene-graph/node.ts [hunk 2]
```diff
@@ -2153,16 +2163,26 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
     public setRotation(x: number, y: number, z: number, w: number): void;
 
     public setRotation (val: Readonly<Quat> | number, y?: number, z?: number, w?: number): void {
+        let x: number;
+        const localRotation = this._lrot;
+
         if (y === undefined || z === undefined || w === undefined) {
```
Comment (dumganhar) [inline cocos/scene-graph/node.ts:2169]: Should it be `and` logic here  like which in setPosition?

Comment (minggo) [reply cocos/scene-graph/node.ts:2169]: I don't want to change its original logic in this PR.

Comment (dumganhar) [reply cocos/scene-graph/node.ts:2169]: Okay.

**Review by dumganhar — APPROVED** [review]

**Pull request merged**